xl/xenops_server_xenlight.ml

```diff
@@ -2256,9 +2256,11 @@ module VM = struct
 						xs.Xs.setperms ent rwperm
 					)
 				) (
-					let dev_kinds = [ "vbd"; "vif"; "vfb"; "vkb"; "vfs"; "pci" ] in
 					[ "device"; "error"; "drivers"; "control"; "attr"; "data"; "messages"; "vm-data" ]
-					@ List.map (fun dev_kind -> "device/"^dev_kind) dev_kinds
+					@ if hvm then
+						let dev_kinds = [ "vbd"; "vif"; "vfb"; "vkb"; "vfs"; "pci" ] in
+						List.map (fun dev_kind -> "device/"^dev_kind) dev_kinds
+					else []
```

Comment:
I know. It's not nice for us to care at this layer. These were added for the Windows update feature and they break old PV guests (specifically the kernel in RHEL4) which takes exception to these paths being created unilaterally. These paths weren't being created at all until a couple of weeks ago.